tests/unit/test_loaders.py

```diff
@@ -36,42 +43,80 @@
 class TestJSONFileLoader(BaseEnvVar):
     def setUp(self):
         super().setUp()
-        self.data_path = os.path.join(os.path.dirname(__file__), 'data')
+        self.data_path = pathlib.Path(__file__).parent.joinpath('data')
+        self.zip_data_path = ZipPath(self.data_path.joinpath('Archive.zip'))
         self.file_loader = JSONFileLoader()
-        self.valid_file_path = os.path.join(self.data_path, 'foo')
-        self.compressed_file_path = os.path.join(self.data_path, 'compressed')
+        self.valid_file_path = self.data_path.joinpath('foo')
+        self.zip_valid_file_path = self.zip_data_path.joinpath('foo')
+        self.compressed_file_path = self.data_path.joinpath('compressed')
+        self.zip_compressed_file_path = self.zip_data_path.joinpath(
+            'compressed'
+        )
 
     def test_load_file(self):
         data = self.file_loader.load_file(self.valid_file_path)
         self.assertEqual(len(data), 3)
         self.assertTrue('test_key_1' in data)
 
+        zip_data = self.file_loader.load_file(self.zip_valid_file_path)
+        self.assertEqual(len(zip_data), 3)
+        self.assertTrue('test_key_1' in zip_data)
+
+        self.assertEqual(data, zip_data)
+
     def test_load_compressed_file(self):
         data = self.file_loader.load_file(self.compressed_file_path)
         self.assertEqual(len(data), 3)
         self.assertTrue('test_key_1' in data)
 
+        zip_data = self.file_loader.load_file(self.zip_compressed_file_path)
+        self.assertEqual(len(zip_data), 3)
+        self.assertTrue('test_key_1' in data)
+
+        self.assertEqual(data, zip_data)
+
     def test_load_compressed_file_exists_check(self):
         self.assertTrue(self.file_loader.exists(self.compressed_file_path))
+        self.assertTrue(self.file_loader.exists(self.zip_compressed_file_path))
 
     def test_load_json_file_does_not_exist_returns_none(self):
         # None is used to indicate that the loader could not find a
         # file to load.
-        self.assertIsNone(self.file_loader.load_file('fooasdfasdfasdf'))
+        self.assertIsNone(
+            self.file_loader.load_file(pathlib.Path('fooasdfasdfasdf'))
```

Comment:
Do we need to make this a pathlib object? Presumably a string should still work with this since it's a public interface.

Reply:
ah I was under the impression the file loader should be treated as a private interface even though it doesn't start with `_`. i'll update it